tests/test-samba-container.sh

```diff
@@ -1,44 +1,71 @@
 #!/bin/bash
 
-LOCAL_TAG="${LOCAL_TAG:-samba-server:latest}"
 
+
+echo "determining container command."
 if [ -z "${CONTAINER_CMD}" ]; then
-	CONTAINER_CMD=$(command -v docker || echo "")
+	CONTAINER_CMD=$(command -v docker || echo )
 fi
 if [ -z "${CONTAINER_CMD}" ]; then
-	CONTAINER_CMD=$(command -v podman || echo "")
+	CONTAINER_CMD=$(command -v podman || echo )
 fi
-
+if [ -z "${CONTAINER_CMD}" ]; then
+echo "Error determining container command."
+exit 1
+fi
+echo "container command: '${CONTAINER_CMD}'."
+echo "creating temporary directory."
 TMPDIR="$(mktemp -d)"
 rc=$?
 
 if [ $rc -ne 0 ]; then
-	echo "Error creating temporary directory"
-else
+	echo "Error creating temporary directory."
+	exit 1
+fi
+	echo "temporary directory: '${TMPDIR}'"
+	echo "starting Samba container."
 	CONTAINER_ID="$(${CONTAINER_CMD} run --network=none --name samba \
 		--volume="${TMPDIR}":/share:Z --rm  -d "${LOCAL_TAG}")"
 	rc=$?
 
 	if [ $rc -ne 0 ]; then
 		echo "Error running samba container"
-	else
-		echo "Container started, ID: '${CONTAINER_ID}'"
+		exit 1
+	fi
+	echo "Container started, ID: '${CONTAINER_ID}'"
 
-		# give samba a second to come up
-		sleep 1
+	# give samba a second to come up
+	sleep 1
 
+	echo "Listing samba shares"
 		${CONTAINER_CMD} exec "${CONTAINER_ID}" smbclient -U% -L 127.0.0.1
 		rc=$?
 
-		if [ $rc -ne 0 ]; then
+		if [ ${rc} -ne 0 ]; then
 			echo "Error listing samba shares"
+			exit 1
 		fi
```

Comment:
We could remove extra indentation(`tab`) starting from LINENO:25 through LINENO:47.

Reply:
@obnoxxx Following my previous comments, as well as @anoopcs9 latest comment, I realized that the problem lays in wrong indentations this script had prior to this PR. I took the liberty to re-format it using helper functions (on-top of your PR). IMHO, it fixes both indentation problems and improves readability: 

https://github.com/samba-in-kubernetes/samba-container/commit/0b8988668570f1a162c848a5847dfa7fcf4424b9

Reply:
> We could remove extra indentation(`tab`) starting from LINENO:25 through LINENO:47.

fixed, thanks!